find_email_addresses.py

```diff
@@ -0,0 +1,183 @@
+from bs4 import BeautifulSoup
+from email.utils import parseaddr
+from urlparse import urlparse
+
+import re
+import requests
+import sys
+
+class Email_Finder():
+	def __init__(self, debug = False, find_all_internal_links = False):
+		# Turns on/off debug mode to print what is happening
+		self.debug = debug
+		# self.find_all_internal_links determines whether the Email Finder should find internal links on all internal pages, or only the homepage
+		# web.mit.edu, for example, is very deep with when searching all internal links
+		self.find_all_internal_links = find_all_internal_links
+		if find_all_internal_links:
+			self.debug_print("Finding all internal links")
+		else:
+			self.debug_print("Finding internal links only from homepage")	
+		self.reset()
+	
+	def reset(self):
+		""" Resets the email Email_Finder
+		"""
+
+		self.debug_print("Resetting Email_Finder")
+		self.all_pages = []
+		self.to_do_pages = []
+		self.all_emails = []
+
+
+	def process_domain(self, domain):
+		""" Processes all the internal links in a domain to find all the email addresses
+		"""
+		
+		self.reset()
+		self.domain = domain
+		# setup the regex used to see if urls are internal
+		self.domain_regex = "//" + self.domain + "|www."+ self.domain
+		
+		self.debug_print("Processing domain: " + domain)
+		#Process the homepage
+		self.process_page("", True)
+	
+		# Goes through all the pages found and processes them
+		while (len(self.to_do_pages) > 0):
+			page = self.to_do_pages.pop()
+			self.process_page(page, False)
+		self.print_emails()
+
+	def build_url(self, path):
+		""" Takes a path and generates the URL. 
+		"""
+
+		# If the path is accidently absolute, get the relative path
+		if self.domain in path:
+			path = path.split(self.domain)[1]
```

Comment:
So, based on everything I saw, this should never happen since links are saved as relative paths. If this is hit, something has changed, an href has a strange, invalid path or urlparse worked incorrectly. A better way of handling this might be to throw an error. I didn't do much error handling in this at all. Would you like me to?